doc/man7/EVP_SIGNATURE-ED25519.pod

```diff
@@ -54,25 +49,32 @@ since version 1.1.1.
 Valid algorithm names are B<ed25519>, B<ed448> and B<eddsa>. If B<eddsa> is
 specified, then both Ed25519 and Ed448 are benchmarked.
 
+
 =head1 EXAMPLES
 
-This example generates an B<ED25519> private key and writes it to standard
-output in PEM format:
+To sign a message using a ED25519 or ED448 key:
+
+    do_sign(EVP_PKEY *ed_key, unsigned char *msg, size_t msg_len)
+    {
+    size_t sig_len;
+    unsigned char *sig = NULL;
+    EVP_MD_CTX *md_ctx = EVP_MD_CTX_new();
+
+    EVP_DigestSignInit(md_ctx, NULL, NULL, NULL, ed_key);
+    /* Calculate the requires size for the signature by passing a NULL buffer */
+    EVP_DigestSign(md_ctx, NULL, &sig_len, msg, msg_len);
+    sig = OPENSSL_zalloc(sig_len);
 
- #include <openssl/evp.h>
- #include <openssl/pem.h>
- ...
- EVP_PKEY *pkey = NULL;
- EVP_PKEY_CTX *pctx = EVP_PKEY_CTX_new_id(EVP_PKEY_ED25519, NULL);
- EVP_PKEY_keygen_init(pctx);
- EVP_PKEY_keygen(pctx, &pkey);
- EVP_PKEY_CTX_free(pctx);
- PEM_write_PrivateKey(stdout, pkey, NULL, NULL, 0, NULL, NULL);
+    EVP_DigestSign(md_ctx, sig, &sig_len, msg, msg_len);
+    ...
+    OPENSSL_free(sig);
+    EVP_MD_CTX_free(md_ctx);
+    }
```

Comment:
The indentation in this example looks horrible on github.  Could it be that there are tabs to be converted to 8 spaces?